spec/rubocop/env_usage_cop_spec.rb

```diff
@@ -0,0 +1,276 @@
+          end
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        module MyApp
+          module Datadog
+            Datadog::DATADOG_ENV['DATADOG_API_KEY']
+          end
+        end
+      RUBY
+    end
+
+    it 'registers an offense for ENV.[]' do
+      expect_offense(<<~RUBY)
+        module Datadog
+          ENV.[]('DATADOG_API_KEY')
+          ^^^^^^^^^^^^^^^^^^^^^^^^^ CustomCops/EnvUsageCop: Avoid direct usage of ENV. Use DATADOG_ENV to access environment variables. See docs/AccessEnvironmentVariables.md for details.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        module Datadog
+          DATADOG_ENV.[]('DATADOG_API_KEY')
+        end
+      RUBY
+    end
+
+    it 'registers an offense for ENV.fetch' do
+      expect_offense(<<~RUBY)
+        module Datadog
+          ENV.fetch('DATADOG_API_KEY')
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ CustomCops/EnvUsageCop: Avoid direct usage of ENV. Use DATADOG_ENV to access environment variables. See docs/AccessEnvironmentVariables.md for details.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        module Datadog
+          DATADOG_ENV.fetch('DATADOG_API_KEY')
+        end
+      RUBY
+    end
+
+    it 'registers an offense for ENV.fetch with default' do
+      expect_offense(<<~RUBY)
+        module Datadog
+          ENV.fetch('DATADOG_API_KEY', 'default')
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ CustomCops/EnvUsageCop: Avoid direct usage of ENV. Use DATADOG_ENV to access environment variables. See docs/AccessEnvironmentVariables.md for details.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        module Datadog
+          DATADOG_ENV.fetch('DATADOG_API_KEY', 'default')
+        end
+      RUBY
+    end
+
+    it 'registers an offense for ENV.fetch with block' do
+      expect_offense(<<~RUBY)
+        module Datadog
+          ENV.fetch('DATADOG_API_KEY') do
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ CustomCops/EnvUsageCop: Avoid direct usage of ENV. Use DATADOG_ENV to access environment variables. See docs/AccessEnvironmentVariables.md for details.
+            'default'
+          end
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        module Datadog
+          DATADOG_ENV.fetch('DATADOG_API_KEY') do
+            'default'
+          end
+        end
+      RUBY
+    end
+
+    it 'registers an offense for ENV.fetch with inline block' do
+      expect_offense(<<~RUBY)
+        module Datadog
+          ENV.fetch('DATADOG_API_KEY') { 'default' }
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ CustomCops/EnvUsageCop: Avoid direct usage of ENV. Use DATADOG_ENV to access environment variables. See docs/AccessEnvironmentVariables.md for details.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        module Datadog
+          DATADOG_ENV.fetch('DATADOG_API_KEY') { 'default' }
+        end
+      RUBY
+    end
```

Comment:
Minor: Up to you if you want to leave all these, but I'm still of the opinion we should have the minimal set of tests that serves our purposes, rather than trying to add a lot of arbitrary cases that should be taken care of by the tool we're using.